lib/signature.js

```diff
@@ -33,37 +33,40 @@ function Signature(opts) {
 }
 
 Signature.prototype.toBuffer = function (format) {
-	if (format === undefined)
-		format = 'asn1';
-	assert.string(format, 'format');
+	if (format !== undefined)
+		assert.string(format, 'format');
 
 	var buf;
 	var stype = 'ssh-' + this.type;
 
 	switch (this.type) {
 	case 'rsa':
-		switch (this.hashAlgorithm) {
-		case 'sha256':
-			stype = 'rsa-sha2-256';
-			break;
-		case 'sha512':
-			stype = 'rsa-sha2-512';
-			break;
-		case 'sha1':
-		case undefined:
-			break;
-		default:
-			throw (new Error('SSH signature ' +
-			    'format does not support hash ' +
-			    'algorithm ' + this.hashAlgorithm));
-		}
 		if (format === 'ssh') {
+			switch (this.hashAlgorithm) {
+			case 'sha256':
+				stype = 'rsa-sha2-256';
+				break;
+			case 'sha512':
+				stype = 'rsa-sha2-512';
+				break;
```

Comment:
Silly question.  Why is there no case for 'sha384'? Pardon any ignorance; is it non-standard/not-supported?  Does the 'sha1'/undefined case cover it?